version: "3.3"

services:

    #####################
    ### --- Magi ---- ###
    #####################

    magi_dev:
        image:
            safetel/safetel-back:dev
        ports:
            - 4270:4270
        restart:
            on-failure
        environment:
            - SERVER_PORT=4270
        env_file: ../dev-config.env

    magi_postman:
        image:
            safetel/safetel-back:postman
        ports:
            - 4207:4207
        restart:
            on-failure
        environment:
            - SERVER_PORT=4207
        env_file: ../postman-config.env

    magi_prod:
        image:
            safetel/safetel-back:prod
        ports:
            - 2407:2407
        restart:
            on-failure
        environment:
            - SERVER_PORT=2407
        env_file: ../prod-config.env

    #####################
    ### --- Amane --- ###
    #####################

    clean_db:
        build: # "context" and "dockerfile" fields have to be under "build"
            context: ../../tools/Amane/
            dockerfile: ./Dockerfile
        restart:
            on-failure
        env_file: ../dev-config.env

    clean_db_dev_force_now:
        build:
            context: ../../tools/Amane/
            dockerfile: ./Dockerfile
        command: ["python3", "./src/Amane.py", "--force-now"]
        restart:
            on-failure
        env_file: ../dev-config.env

    clean_db_postman_force_now:
        build:
            context: ../../tools/Amane/
            dockerfile: ./Dockerfile
        command: ["python3", "./src/Amane.py", "--force-now"]
        restart:
            on-failure
        env_file: ../postman-config.env